modules/reactor/doc/content/source/meshgenerators/PatternedHexMeshGenerator.md

```diff
@@ -0,0 +1,50 @@
+
+- `inputs`: a vector of mesh generator names that will be used as elements to form the patterned mesh;
+- `pattern`: a two-dimensional vector that represents the pattern of mesh to be generated. The elements must be integers from 0 to N-1, where N is the length of vector parameter `inputs`.
+- `pattern_boundary`: a MooseEnum parameter that can be either `hexagon` or `none`. When `none` is selected, this object only stitches `inputs` meshes into a patterned mesh without extraneous geometry, resulting in an outer boundary with a zig-zag edge. When `hexagon` is selected, a background region is added so that the generated mesh has a hexagonal shape instead of a zig-zag one. Concentric duct regions may also be optionally added to the hexagon periphery when this parameter is `hexagon`. The user can refer to [Figure 1](#pattern_hex) for more details.
+
+!media media/meshgenerators/pattern_hex.png
+      style=display: block;margin-left:auto;margin-right:auto;width:40%;
+      id=pattern_hex
+      caption=A schematic showing the difference between `none` and `hexagon` `pattern_boundary`.
+
+When `pattern_boundary` is set as `hexagon`, the user can also provide `duct_sizes` in either `apothem` or `radius` style to add external duct regions to the generated hexagon mesh. Also, `hexagon_size` must also be provided to define the external hexagon boundary size, which can be either `apothem` or `radius` of the hexagon, as determined by `hexagon_size_style`. In some cases, if `hexagon_size` is small enough, the hexagon external boundary may cut off some of the stitched hexagonal meshes. As long as the concentric circular (`rings`) regions are not cut off, the rest of the mesh is deformed to accommodate the limited external boundaries.
+
+## Control Drum Related MeshMetaData
+
+One of the applications of this object is to generate meshes for prismatic reactor cores. In that case, by setting `generate_core_metadata` as true, control drum meshes can also be used as part of `inputs` to construct the core mesh. To facilitate the use of control drum rotation simulation objects, a series of `MeshMetaData` can be generated, including:
+
+- `control_drum_positions`: a vector of control drum center positions. This `MeshMetaData` can also be outputted as an ASCII file by setting `generate_control_drum_positions_file` as true and providing `position_file`;
+- `control_drum_angles`: a vector of the azimuthal angles of the control drum center positions to the center of the core.
+- `control_drums_azimuthal_meta`: a two-dimensional vector containing the sorted azimuthal angles of nodes to the corresponding control drum center for all the control drums.
+
+In addition, `assign_control_drum_id` can be set as true so that the control drum `inputs` meshes can be indexed using an element extra integer called `control_drum_id`. As illustrated in [Figure 2](#cd_id), the `control_drum_id` is indexed based on the azimuthal angles of the control drums.
```

Comment:
```suggestion
In addition, [!param](/Mesh/PatternedHexMeshGenerator/assign_control_drum_id) can be set as true so that the control drum [!param](/Mesh/PatternedHexMeshGenerator/inputs) meshes can be indexed using an element extra integer called `control_drum_id`. As illustrated in [Figure 2](#cd_id), the `control_drum_id` is indexed based on the azimuthal angles of the control drums.
```